compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/diagnostics/FirDefaultErrorMessages.kt

```diff
@@ -570,6 +573,15 @@ class FirDefaultErrorMessages {
             map.put(CLASS_IN_SUPERTYPE_FOR_ENUM, "Enum class cannot inherit from classes")
             map.put(SEALED_SUPERTYPE, "This type is sealed, so it can be inherited by only its own nested classes or objects")
             map.put(SEALED_SUPERTYPE_IN_LOCAL_CLASS, "Local class cannot extend a sealed class")
+            map.put(
+                SEALED_INHERITOR_IN_DIFFERENT_PACKAGE,
+                "Inheritor of sealed class or interface declared in package {0} but it must be in package {1} where base class is declared",
```

Comment:
I was trying to see if the package names were in the right order in the message. But I found that the structure/grammar of this message is confusing and ambiguous. `{0}` could easily refer to the package of the sealed class or interface, and not the inheritor. Similarly, `{1}` could be interpreted as the package where the base class _is_ declared, not where it _should_ be.

I don't know if we are ok to change these messages, but I would even simplify it to `"A type can only inherit from a sealed class or interface if they are in the same package"`. Or if we absolutely must keep the package names: `"This supertype is a sealed class or interface in package {0} but the inheritor (currently in package {1}) must be in the same package"` and switch the package names around.

(Also "java" in the message below should be capitalized! But I see in FE 1.0 it is not.)

Reply:
OK, let me rephrase the message and put it into a separate commit so it they can opt to merge this separate commit or not. BTW, an interface cannot be sealed so I will drop that part.

Reply:
Looks good, but you should keep the interface part. Java has sealed interfaces (see `compiler/testData/diagnostics/tests/testsWithJava17/sealedClasses/kotlinInheritsJavaInterface.kt`). I just learned about this today too. 🤯

Reply:
I don't think sealed Java interface is relevant here because it's an error to implement a sealed Java interface in Kotlin, which is why we have the `CLASS_INHERITS_JAVA_SEALED_CLASS` diagnostic devoted for this case.